pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1554 @@
+			}
+
+			k.Responses[v.Name] = p
+		}
+	}
+
+	if g.SecurityDefinitions != nil {
+		k.SecurityDefinitions = make(SecurityDefinitions)
+		if nok, err := k.SecurityDefinitions.FromGnostic(g.SecurityDefinitions); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	if g.Security != nil {
+		k.Security = make([]map[string][]string, len(g.Security))
+		for i, v := range g.Security {
+			if v.AdditionalProperties == nil {
+				continue
+			}
+
+			k.Security[i] = make(map[string][]string, len(v.AdditionalProperties))
+			converted := k.Security[i]
+			for _, p := range v.AdditionalProperties {
+				if p.Value != nil {
+					converted[p.Name] = p.Value.Value
+				} else {
+					converted[p.Name] = nil
+				}
+			}
+		}
+	}
+
+	if g.Tags != nil {
+		k.Tags = make([]Tag, len(g.Tags))
+		for i, v := range g.Tags {
+			if v == nil {
+				continue
+			} else if nok, err := k.Tags[i].FromGnostic(v); err != nil {
+				return false, err
+			} else if !nok {
+				ok = false
+			}
+		}
+	}
+
+	if g.ExternalDocs != nil {
+		k.ExternalDocs = &ExternalDocumentation{}
+		if nok, err := k.ExternalDocs.FromGnostic(g.ExternalDocs); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	return ok, nil
+}
+
+// Info
+
+func (k *Info) FromGnostic(g *openapi_v2.Info) (ok bool, err error) {
+	ok = true
+	if g == nil {
+		return true, nil
+	}
+
+	if nok, err := k.VendorExtensible.FromGnostic(g.VendorExtension); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	if nok, err := k.InfoProps.FromGnostic(g); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	return ok, nil
+}
+
+func (k *InfoProps) FromGnostic(g *openapi_v2.Info) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	ok = true
+
+	k.Description = g.Description
+	k.Title = g.Title
+	k.TermsOfService = g.TermsOfService
+
+	if g.Contact != nil {
+		k.Contact = &ContactInfo{}
+
+		if nok, err := k.Contact.FromGnostic(g.Contact); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	if g.License != nil {
+		k.License = &License{}
+		if nok, err := k.License.FromGnostic(g.License); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	k.Version = g.Version
+	return ok, nil
+}
+
+func (k *License) FromGnostic(g *openapi_v2.License) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	ok = true
+
+	k.Name = g.Name
+	k.URL = g.Url
+
+	// License does not embed to VendorExtensible!
+	// data loss from g.VendorExtension
+	if len(g.VendorExtension) != 0 {
+		ok = false
+	}
+
+	return ok, nil
+}
+
+func (k *ContactInfo) FromGnostic(g *openapi_v2.Contact) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	ok = true
+
+	k.Name = g.Name
+	k.URL = g.Url
+	k.Email = g.Email
+
+	// ContactInfo does not embed to VendorExtensible!
+	// data loss from g.VendorExtension
+	if len(g.VendorExtension) != 0 {
+		ok = false
```

Comment:
this data loss code path is missing test coverage

Reply:
added a specific test targeting data loss cases